File: public/usage-examples/animations/assign_animation_with_sound-1-example.cpp
#include "splashkit.h"

int main()
{
    animation_script script = load_animation_script("WalkingScript", "kermit.txt");
    animation anim = create_animation(script, "WalkFront");

    // Later, switch to WalkFront and optionally play its starting-frame sound
    assign_animation(anim, "WalkFront", true);
    write_line("Assigned WalkFront with sound -> animation name: %s", animation_name(anim).c_str());

    free_animation(anim);
    free_animation_script(script);
    return 0;
}

File: public/usage-examples/animations/assign_animation_with_sound-1-example.py
from splashkit import *

script = load_animation_script("WalkingScript", "kermit.txt")
anim = create_animation(script, "WalkFront")

# Assign by name and play sound if present on start frame
assign_animation_with_sound(anim, "WalkFront", True)
write_line(f"Assigned WalkFront with sound -> animation name: {animation_name(anim)}")

free_animation(anim)
free_animation_script(script)
